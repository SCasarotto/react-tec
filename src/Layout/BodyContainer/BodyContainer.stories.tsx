import React from 'react';

import { Story, Meta } from '@storybook/react';

import { BodyContainer, BodyContainerProps } from './BodyContainer';

export default {
  title: 'Layout/BodyContainer',
  component: BodyContainer,
  argTypes: {},
} as Meta;

const Template: Story<BodyContainerProps> = (args) => (
  <BodyContainer {...args} />
);

export const Default = Template.bind({});
Default.args = {};
