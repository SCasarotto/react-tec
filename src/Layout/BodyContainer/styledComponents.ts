import styled from 'styled-components';

export const StyledBodyContainer = styled.div<{ sidebarWidth?: number }>`
  position: relative;
  width: calc(100% - ${(props) => props.sidebarWidth}px);
  min-height: 100vh;
  margin-left: ${(props) => props.sidebarWidth}px;
  background-color: ${(props) => props.theme.lighterGray};
`;
