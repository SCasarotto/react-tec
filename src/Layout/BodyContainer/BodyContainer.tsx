import React from 'react';

import { StyledBodyContainer } from './styledComponents';

export interface BodyContainerProps
  extends React.HTMLAttributes<HTMLDivElement> {
  sidebarWidth?: number;
}
/**
 *
 * `BodyContainer` is really just a sized meant to play well with `SideNavbar`. Below is the exact code used in this site.
 *
 * ```
 *	<BodyContainer>
 *		<TitleBar title={barTitle} />
 *		{children}
 *	</BodyContainer>
 * ```
 *
 */
export const BodyContainer: React.FC<BodyContainerProps> = (props) => {
  const { className = '', ...rest } = props;
  return (
    <StyledBodyContainer
      className={`tec-bodyContainer ${className}`}
      {...rest}
    />
  );
};

//TODO: Not in love with this. It might be a good idea to find a better way while also introducing some default responsiveness
BodyContainer.defaultProps = {
  sidebarWidth: 200,
};
