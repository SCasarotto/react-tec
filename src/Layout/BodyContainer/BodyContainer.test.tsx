import React from 'react';

import { render } from '@testing-library/react';

import { BodyContainer } from './BodyContainer';

describe('BodyContainer', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<BodyContainer />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<BodyContainer className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
