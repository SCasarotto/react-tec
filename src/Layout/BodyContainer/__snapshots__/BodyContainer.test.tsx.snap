// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyContainer matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  width: calc(100% - 200px);
  min-height: 100vh;
  margin-left: 200px;
}

<div
    class="c0 tec-bodyContainer "
  />
</DocumentFragment>
`;
